eslint-plugin-expensify/prefer-at.js

```diff
@@ -0,0 +1,116 @@
+                node.property.name === 'length'
+            );
+        }
+
+        function getSourceCode(node) {
+            return context.getSourceCode().getText(node);
+        }
+
+        function checkNode(node) {
+            if (node.type === 'MemberExpression' && node.property) {
+                // Check if the item being accessed is an array
+                if (!isArrayType(node.object)) {
+                    return;
+                }
+
+                // Handle items like test[0] and array literals like [0, 1, 2, 3, 4][1]
+                if (isLiteralNumber(node.property)) {
+                    context.report({
+                        node,
+                        message,
+                        fix(fixer) {
+                            const objectText = getSourceCode(node.object);
+                            return fixer.replaceText(node, `${objectText}.at(${node.property.value})`);
+                        },
+                    });
+                }
+
+                // Handle items like test[test.length - 1]
+                if (
+                    node.property.type === 'BinaryExpression' &&
+                    node.property.operator === '-' &&
+                    isMemberExpressionWithLength(node.property.left) &&
+                    isLiteralNumber(node.property.right)
+                ) {
+                    context.report({
+                        node,
+                        message,
+                        fix(fixer) {
+                            const objectText = getSourceCode(node.object);
+                            const offset = node.property.right.value;
+                            return fixer.replaceText(node, `${objectText}.at(-${offset})`);
+                        },
+                    });
+                }
+
+                // Handle items like test[(test.length - 1) / 2]
+                if (
+                    node.property.type === 'BinaryExpression' &&
+                    node.property.operator === '/' &&
+                    node.property.left.type === 'BinaryExpression' &&
+                    node.property.left.operator === '-' &&
+                    isMemberExpressionWithLength(node.property.left.left) &&
+                    isLiteralNumber(node.property.left.right) &&
+                    isLiteralNumber(node.property.right)
+                ) {
+                    context.report({
+                        node,
+                        message,
+                        fix(fixer) {
+                            const objectText = getSourceCode(node.object);
+                            const leftValue = node.property.left.right.value;
+                            const rightValue = node.property.right.value;
+                            return fixer.replaceText(node, `${objectText}.at((${objectText}.length - ${leftValue}) / ${rightValue})`);
+                        },
+                    });
+                }
```

Comment:
Is possible to make this code more generic to actually capture anything inside `[...]` and replace with `.at(...)` if applicable? So in this way we don't have to rely in super-specific use cases like these ones and we instead will cover all possibilities.